backend/src/apiserver/server/job_server.go

```diff
@@ -149,21 +164,47 @@ func (s *JobServer) validateCreateJobRequest(request *api.CreateJobRequest) erro
 	}
 
 	if job.MaxConcurrency > 10 || job.MaxConcurrency < 1 {
-		return util.NewInvalidInputError("The max concurrency of the job is out of range. Support 1-10. Received %v.", job.MaxConcurrency)
+		return util.NewInvalidInputError("supported max concurrency in range [1, 10], got %v.", job.MaxConcurrency)
 	}
-	if job.Trigger != nil && job.Trigger.GetCronSchedule() != nil {
-		if _, err := cron.Parse(job.Trigger.GetCronSchedule().Cron); err != nil {
+
+	return s.validateTrigger(job.Trigger)
+}
+
+func (s *JobServer) validateJobUpdate(job *api.Job) error {
+	if job.Name == "" {
+		return util.NewInvalidInputError("job name missing")
+	}
+
+	if job.GetPipelineSpec() == nil {
+		return util.NewInvalidInputError("pipeline spec missing")
+	}
+
+	if job.MaxConcurrency > 10 || job.MaxConcurrency < 1 {
+		return util.NewInvalidInputError("supported max concurrency in range [1, 10], got %v.", job.MaxConcurrency)
+	}
+
+	if job.Trigger == nil {
+		return util.NewInvalidInputError("trigger missing")
+	}
+	return s.validateTrigger(job.Trigger)
+}
+
+func (s *JobServer) validateTrigger(trigger *api.Trigger) error {
+	if trigger.GetCronSchedule() != nil {
```

Comment:
Seems that the passed-in "trigger" could be nil. Please use trigger != nil && trigger.GetCronSchedule() != nil.